File: 057_very_hard_Find_the_Pattern_and_Write_the_Function.go
/*
Find the Pattern and Write the Function
By looking at the inputs and outputs below, try to figure out the pattern and write a function to execute it for any number.

Examples
func(3456) ➞ 2

func(89265) ➞ 5

func(97) ➞ 12

func(2113) ➞ -9"

*/

package main

import (
	"fmt"
	"strconv"
)

func funcDigits(n int) int {
	s := strconv.Itoa(n)

	sum := 0
	for _, ch := range s {
		sum += int(ch - '0')
	}

	k := len(s)
	return sum - k*k
}
func main() {
	// テスト出力
	fmt.Println(funcDigits(3456))   // 2
	fmt.Println(funcDigits(89265))  // 5
	fmt.Println(funcDigits(97))     // 12
	fmt.Println(funcDigits(2113))   // -9
	fmt.Println(funcDigits(55))     // 6
	fmt.Println(funcDigits(785428)) // -2
	fmt.Println(funcDigits(439))    // 7
	fmt.Println(funcDigits(55654))  // 0
}

File: 057_very_hard_Find_the_Pattern_and_Write_the_Function.py
"""Find the Pattern and Write the Function
By looking at the inputs and outputs below, try to figure out the pattern and write a function to execute it for any number.

Examples
func(3456) ➞ 2

func(89265) ➞ 5

func(97) ➞ 12

func(2113) ➞ -9"
"""

def func(num: int) -> int:
    
    digits = str(num)
    return sum(map(int, digits)) - len(digits) ** 2
    
    # sum_num = 0
    # for i in list(str(num)):
    #     sum_num += int(i)
    
    # return sum_num - len(str(num)) * len(str(num))


print(func(3456)), # 2)
print(func(89265)), # 5)
print(func(97)), # 12)
print(func(2113)), # -9)
print(func(55)), # 6)
print(func(785428)), # -2)
print(func(439)), # 7)
print(func(55654)), # 0)